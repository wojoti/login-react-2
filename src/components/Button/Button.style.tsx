import styled from "styled-components";
export const Wrapper = styled.input`
  width: 100%;
  height: 32px;
  border-radius: 6px;
  border: 0px solid black;
  --tw-bg-opacity: 1;
  background-color: rgb(244 63 94 / var(--tw-bg-opacity));
  color: white;
  cursor: pointer;
`;
