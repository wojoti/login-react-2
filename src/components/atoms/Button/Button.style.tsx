import styled from "styled-components";
export const StyledButton = styled.input`
  width: 100%;
  height: 32px;
  border-radius: 6px;
  border: 0px solid black;
  --tw-bg-opacity: 1;
  background-color: rgb(244 63 94 / var(--tw-bg-opacity));
  color: white;
  cursor: pointer;
  --tw-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
  --tw-shadow-colored: 0 25px 50px -12px var(--tw-shadow-color);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),
    var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
`;
