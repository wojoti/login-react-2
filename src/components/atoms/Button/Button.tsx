import { forwardRef } from "react";
import StyledButton from "./Button.style";

export interface ButtonProps {
  name: string;
  onClick: () => void;
  testId?: string;
  disabled?: boolean;
}

export type ButtonRef = HTMLButtonElement;

const Button = forwardRef<ButtonRef, ButtonProps>(
  ({ name, onClick, testId, disabled = false }, ref) => (
    <StyledButton
      type="submit"
      onClick={onClick}
      data-testid={testId}
      ref={ref}
      disabled={disabled}
    >
      {name}
    </StyledButton>
  )
);
Button.displayName = "Button";

export default Button;
