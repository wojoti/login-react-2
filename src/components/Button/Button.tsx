import React from "react";
import { Wrapper } from "./Button.style";

type Props = {
  name?: string;
  id?: string;
  children?: string;
  onSubmit?: (e: any) => void;
};

const Button = (props: Props) => {
  return (
    <Wrapper
      type="submit"
      name={props.name}
      id={props.id}
      value={props.children}
      onClick={props.onSubmit}
    ></Wrapper>
  );
};

export default Button;
