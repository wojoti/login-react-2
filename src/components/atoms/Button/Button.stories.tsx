/* eslint-disable import/no-extraneous-dependencies */
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Button from "./Button";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Atoms/Button",
  component: Button,
} as ComponentMeta<typeof Button>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
// eslint-disable-next-line react/jsx-props-no-spreading, react/function-component-definition
const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Login = Template.bind({});
Login.args = {
  name: "LOGIN",
};

export const SignUp = Template.bind({});
SignUp.args = {
  name: "SIGN UP",
};
