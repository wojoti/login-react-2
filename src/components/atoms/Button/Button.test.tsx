import { fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import Button, { ButtonProps } from "./Button";

const handleClick = jest.fn();
const props: ButtonProps = {
  name: "test button",
  onClick: handleClick,
  testId: "test-button-id",
};

const modifiedProps: ButtonProps = {
  ...props,
  disabled: true,
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<Button {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render matching button with text", () => {
  render(<Button {...props} />);
  const buttonElement = screen.getByTestId("test-button-id");
  expect(buttonElement).toBeInTheDocument();
  expect(buttonElement).toHaveTextContent("test button");
});

test("should handle onClick event", () => {
  render(<Button {...props} />);
  const buttonElement = screen.getByTestId("test-button-id");
  fireEvent.click(buttonElement);
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test("shouldn't be able to handle onClick when disabled", () => {
  render(<Button {...modifiedProps} />);
  const buttonElement = screen.getByTestId("test-button-id");
  fireEvent.click(buttonElement);
  expect(handleClick).toHaveBeenCalledTimes(0);
});
