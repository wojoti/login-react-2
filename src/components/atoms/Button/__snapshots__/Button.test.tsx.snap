// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
  height: 32px;
  border-radius: 6px;
  border: 0px solid black;
  background-color: rgb(244 63 94);
  background-color: rgb(244 63 94);
  color: white;
  cursor: pointer;
  box-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
}

<button
  className="c0"
  data-testid="test-button-id"
  disabled={false}
  onClick={[MockFunction]}
  type="submit"
>
  test button
</button>
`;
